### DISCUSSION TITLE
{admiral} date-time cyclomatic complexity cleanup

### DISCUSSION THREAD

**Discussion by zdz2101** [pharmaverse/admiral#2332] [Ideas]
We are seeking to reduce the cyclomatic complexity of the date/datetime functions in our codebase, e.g.`derive_vars_dtm()`

@pharmaverse/admiral please comment here if you'd like to be involved.

**Comment by bms63** [comment]
I'm in for this one.  Should be really interesting!!!

**Reply by millerg23** [reply]
count me in.

**Reply by zdz2101** [reply]
Looks like we can revive the old-lab working group! I'll slack yall about meeting times for a kick off, I've created an issue to start "demo"-ing some of the tasks that can be done to start tackling this daunting task

**Reply by zdz2101** [reply]
Mural to sketch of mindmap of how to break it down:

https://app.mural.co/t/admiraldevelopmentteam6136/m/admiraldevelopmentteam6136/1708018668425/50fce14599531b503b2f568061c1f274482c47f8?sender=u2186aa07ff6a351a07a19987

**Reply by bms63** [reply]
Looks awesome!  Maybe we can convert some of the test files to help us walk through each step of the process?

One thing is we should really up our unit test game before we dive too deep into re-working the functions

**Reply by StefanThoma** [reply]
Hey all, I'd be interested to join this effort as well.

**Comment by zdz2101** [comment]
Created issues pertaining to smaller bite sized chunks

- [ ] #2342
- [ ] #2339
- [ ] #2340
- [ ] #2343
- [ ] Add unit tests to each function and add edge cases to ensure our functions are doing what we expect (to be parsed into maybe 7-8 issues)
- [ ] Potentially collapse some of the embedded functions/wrappers so that `derive_` function does not go 4-layers deep, wonder if `impute_dtc_xxx()` and `restricted_imputed_dtc_xxx()` can be more "intuitive" (tentative)

**Comment by zdz2101** [comment]
Continue talk about the `preserve` argument and the exotic ways to use these functions

**Comment by manciniedoardo** [comment]
@zdz2101 do you know where we are with this?

**Comment by manciniedoardo** [comment]
Closing as this has been explored.
